src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,46 @@
 
 class Animal {
   // write your code here
+
+  static alive = [];
+
+  constructor(name) {
+    this.health = 100;
+    this.name = name;
+    Animal.alive.push(this);
+  }
+
+  static checkAlive() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name, hidden = false) {
+    super(name);
+    this.hidden = hidden;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+
+  bite(target) {
+    if (target instanceof Carnivore) {
+      return;
+    }
+
+    if (target instanceof Herbivore && !target.hidden) {
+      target.health -= 50;
+
+      Animal.checkAlive();
+    }
```

Comment:
The `bite` method correctly checks if the target is a Carnivore and exits if so. However, when checking if the target is a Herbivore, it should also check if the `hidden` property is `false` before applying damage. After decreasing the health of the herbivore, it's good to call `Animal.checkAlive` to update the `Animal.alive` array, but make sure that this method is efficient and does not lead to performance issues with nested loops.